backend/operations_scanner.go

```diff
@@ -147,11 +226,19 @@ loop:
 // Join waits for the OperationsScanner to gracefully shut down.
 func (s *OperationsScanner) Join() {
 	s.subscriptionWorkers.Wait()
+	s.leaderGauge.Set(0)
 }
 
 // collectSubscriptions builds an internal list of Azure subscription IDs by
 // querying Cosmos DB.
 func (s *OperationsScanner) collectSubscriptions(ctx context.Context, logger *slog.Logger) {
+	startTime := time.Now()
+	s.operationsCount.WithLabelValues(collectSubscriptionsLabel).Inc()
+	defer func() {
+		s.operationsDuration.WithLabelValues(collectSubscriptionsLabel).Observe(time.Since(startTime).Seconds())
+		s.lastOperationTimestamp.WithLabelValues(collectSubscriptionsLabel).SetToCurrentTime()
+	}()
```

Comment:
Would it work to make this anonymous function a proper method so it doesn't have to keep being repeated?  Something like...
```
func (s *OperationsScanner) updateOperationMetrics(label string, startTime time.Time)
```
and then here you could call
```
defer s.updateOperationMetrics(collectSubscriptionsLabel, startTime)
```

Reply:
Honestly what I really wish is golang had the concept of decorators like in Python, because this would be a perfect use case for it.

Reply:
good idea! I've added a slight modification to your suggestion.